src/js/main.js

```diff
@@ -707,95 +685,93 @@ function updateTabList(features) {
 }
 
 function zeroPad(value, width) {
-    value = "" + value;
 
-    while (value.length < width) {
-        value = "0" + value;
+    let valuePadded = String(value);
+
+    while (valuePadded.length < width) {
+        valuePadded = `0${value}`;
     }
 
-    return value;
+    return valuePadded;
 }
 
 function generateFilename(prefix, suffix) {
-    var date = new Date();
-    var filename = prefix;
+    const date = new Date();
+    let filename = prefix;
 
     if (CONFIG) {
         if (CONFIG.flightControllerIdentifier) {
-            filename = CONFIG.flightControllerIdentifier + '_' + filename;
+            filename = `${CONFIG.flightControllerIdentifier}_${filename}`;
         }
         if(CONFIG.name && CONFIG.name.trim() !== '') {
-            filename = filename + '_' + CONFIG.name.trim().replace(' ', '_');
+            filename = `${filename}_${CONFIG.name.trim().replace(' ', '_')}`;
         }
     }
 
-    filename = filename + '_' + date.getFullYear()
-        + zeroPad(date.getMonth() + 1, 2)
-        + zeroPad(date.getDate(), 2)
-        + '_' + zeroPad(date.getHours(), 2)
-        + zeroPad(date.getMinutes(), 2)
-        + zeroPad(date.getSeconds(), 2);
+    const yyyymmdd = `${date.getFullYear()}${zeroPad(date.getMonth() + 1, 2)}${zeroPad(date.getDate(), 2)}`;
+    const hhmmss = `${zeroPad(date.getHours(), 2)}${zeroPad(date.getMinutes(), 2)}${zeroPad(date.getSeconds(), 2)}`;
+    filename = `${filename}_${yyyymmdd}_${hhmmss}`;
 
-    return filename + '.' + suffix;
+    return `${filename}.${suffix}`;
 }
 
 function getTargetVersion(hardwareId) {
-    var versionText = '';
+    let versionText = '';
 
     if (hardwareId) {
-       versionText += i18n.getMessage('versionLabelTarget') + ': ' + hardwareId;
+       versionText = `${i18n.getMessage('versionLabelTarget')}: ${hardwareId}`;
     }
 
     return versionText;
 }
 
 function getFirmwareVersion(firmwareVersion, firmwareId) {
-    var versionText = '';
+    let versionText = '';
 
     if (firmwareVersion) {
-        versionText += i18n.getMessage('versionLabelFirmware') + ': ' + firmwareId + ' ' + firmwareVersion;
+        versionText = `${i18n.getMessage('versionLabelFirmware')}: ${firmwareId} ${firmwareVersion}`;
     }
 
     return versionText;
 }
 
 function getConfiguratorVersion() {
-    return i18n.getMessage('versionLabelConfigurator') + ': ' + CONFIGURATOR.version;
+    return `${i18n.getMessage('versionLabelConfigurator')}: ${CONFIGURATOR.version}`;
 }
 
 function updateTopBarVersion(firmwareVersion, firmwareId, hardwareId) {
-    var versionText = getConfiguratorVersion() + '<br />';
-
-    versionText = versionText + getFirmwareVersion(firmwareVersion, firmwareId) + '<br />';
-
-    versionText = versionText + getTargetVersion(hardwareId);
+    const versionText = String(getConfiguratorVersion())
+                                .concat('<br />')
```

Comment:
Could probably be done in a more readable way with a template string.
Also, indentation is wrong.

Reply:
I tried with string template and in this case seemed to me more complicated. But I will change it and we will see.